travis_ci/test.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+#
+# This file is part of the KubeVirt project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2019 Red Hat, Inc.
+#
+
+name=$1
+workload=$2
+
+travis_fold_start(){
+    echo -e "travis_fold:start:details\033[33;1mDetails\033[0m"
+}
+
+travis_fold_end(){
+    echo -e "\ntravis_fold:end:details\r"
+}
+
+# Generate templates
+make generate
+# Limit required memory of large templates
+bash automation/x-limit-ram-size.sh
+
+# Download images
+case "$name" in
+"fedora")
+	curl -fL -o "$name" https://download.fedoraproject.org/pub/fedora/linux/releases/30/Cloud/x86_64/images/Fedora-Cloud-Base-30-1.2.x86_64.qcow2
+    ;;
+"ubuntu")
+	curl -fL -o "$name" http://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-amd64.img
+    ;;
+"opensuse")
+	curl -fL -o "$name" https://download.opensuse.org/repositories/Cloud:/Images:/Leap_15.0/images/openSUSE-Leap-15.0-OpenStack.x86_64-0.0.4-Buildlp150.12.12.qcow2
+    ;;
+"centos7")
+	curl -fL http://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud.qcow2.xz | xz -d > "$name"
+    ;;
+"centos6")
+	curl -fL http://cloud.centos.org/centos/6/images/CentOS-6-x86_64-GenericCloud.qcow2.xz | xz -d > "$name"
+    ;;
+"centos8")
+# For now we test the CentOS 8 image using Fedora as that is the branch source
+# TODO fix this once CentOS 8 is released
+	curl -fL -o "$name" https://download.fedoraproject.org/pub/fedora/linux/releases/30/Cloud/x86_64/images/Fedora-Cloud-Base-30-1.2.x86_64.qcow2
+    ;;
+esac
+
+# Prepare image
+mkdir -p "$PWD/pvs/$name"
+qemu-img convert -p -O raw $name "$PWD/pvs/$name/disk.img"
+sudo chown 107:107 "$PWD/pvs/$name/disk.img"
+sudo chmod -R a+X "$PWD/pvs"
+
+size_MB=$(( $(qemu-img info $name --output json | jq '.["virtual-size"]') / 1024 / 1024 + 128 ))
+# Create PV and PVC
+bash create-minikube-pvc.sh "$name" "${size_MB}M" "$PWD/pvs/$name/" | tee | oc apply -f -
+
+sizes=("tiny" "small" "medium" "large")
```

Comment:
Not all OSes have to have all sizes. We moved the complicated ones to stdci though (Windows).